pmd-core/src/main/java/net/sourceforge/pmd/reporting/CloseHookFileListener.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.reporting;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import net.sourceforge.pmd.reporting.Report.ProcessingError;
+import net.sourceforge.pmd.reporting.Report.SuppressedViolation;
+
+/**
+ * A delegating listener wrapper that can override the closing behavior.
+ * @param <T> Type of the delegate
+ */
+public abstract class CloseHookFileListener<T extends FileAnalysisListener> implements FileAnalysisListener {
+    private final T delegate;
+
+    public CloseHookFileListener(T delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void onRuleViolation(RuleViolation violation) {
+        delegate.onRuleViolation(violation);
+    }
+
+    @Override
+    public void onSuppressedRuleViolation(SuppressedViolation violation) {
+        delegate.onSuppressedRuleViolation(violation);
+    }
+
+    @Override
+    public void onError(ProcessingError error) {
+        delegate.onError(error);
+    }
+
+    @Override
+    public final void close() throws Exception {
+        Exception e = null;
+        try {
+            delegate.close();
+        } catch (Exception ex) {
```

Comment:
```suggestion
        } catch (Exception _ex) {
```
better mask it here once, for sake of the remaining code.